aiservices/openai/src/main/java/com/microsoft/semantickernel/aiservices/openai/chatcompletion/OpenAIChatCompletion.java

```diff
@@ -497,14 +519,14 @@ private Mono<ChatMessages> internalChatMessageContentsAsync(
                     .flatMap(it -> it)
                     .flatMap(msgs -> {
                         return internalChatMessageContentsAsync(msgs, kernel, functions,
-                            invocationContext, autoInvokeAttempts - 1);
+                            invocationContext, requestIndex + 1);
                     })
                     .onErrorResume(e -> {
 
                         LOGGER.warn("Tool invocation attempt failed: ", e);
 
                         // If FunctionInvocationError occurred and there are still attempts left, retry, else exit
-                        if (autoInvokeAttempts > 0) {
+                        if (requestIndex > 0) {
```

Comment:
is > still the correct logic here now the number is going up?

Reply:
Ah, good catch. It is not